Streetcode/Streetcode.WebApi/Utils/WebParsingUtils.cs

```diff
@@ -247,158 +256,165 @@ await _repository.ToponymRepository.CreateAsync(new Toponym()
+                return (streetname.Substring(streetname.IndexOf(" ") + 1), "більвар");
             }
-            else if (streetname.IndexOf("в’їзд ") != -1)
+            else if (streetname.IndexOf("шосе ", StringComparison.Ordinal) != -1)
             {
-                streetNameStringSearchOptimized = new Tuple<string?, string?>(streetname.Substring(streetname.IndexOf(" ") + 1), "в’їзд");
+                return (streetname.Substring(streetname.IndexOf(" ") + 1), "шосе");
             }
-
-            return streetNameStringSearchOptimized;
-        }
-
-        // Following method calls OSM Nominatim API for single address and returns lat and lon coordinates
-        // For some addresses nothing can be returned
-        public static async Task<Tuple<string, string>?> CallAPICoords(string address)
-        {
-            try
+            else if (streetname.IndexOf("містечко ", StringComparison.Ordinal) != -1)
             {
-                WebClient webClient = new WebClient();
-                webClient.Headers.Add("user-agent", "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.2; .NET CLR 1.0.3705;)");
-                webClient.Headers.Add("Referer", "http://www.microsoft.com");
-                var jsonData = webClient.DownloadData("https://nominatim.openstreetmap.org/search?q=" + address + "&format=json&limit=1&addressdetails=1");
-                string jsonStr = Encoding.UTF8.GetString(jsonData);
-                Tuple<string, string> coords = ParseJsonToCoordinateTuple(jsonStr);
-                return coords;
+                return (streetname.Substring(streetname.IndexOf(" ") + 1), "містечко");
             }
-            catch (Exception ex)
+            else if (streetname.IndexOf("в’їзд ", StringComparison.Ordinal) != -1)
             {
-                Console.ForegroundColor = ConsoleColor.Red;
-                Console.WriteLine(ex.Message);
-                Console.ForegroundColor = ConsoleColor.White;
+                return (streetname.Substring(streetname.IndexOf(" ") + 1), "в’їзд");
             }
 
-            return null;
+            return (null, null);
         }
 
         // Following method parses JSON from OSM Nominatim API and returns lat/lon tuple
-        public static Tuple<string, string> ParseJsonToCoordinateTuple(string json)
+        private static (string, string) ParseJsonToCoordinateTuple(string json)
         {
             var data = JsonConvert.DeserializeObject<List<Dictionary<string, object>>>(json);
             if (data.Count == 0)
             {
-                return new Tuple<string, string>(string.Empty, string.Empty);
+                return (string.Empty, string.Empty);
             }
             else
             {
-                return new Tuple<string, string>(data[0]["lat"].ToString(), data[0]["lon"].ToString());
+                return (data[0]["lat"].ToString(), data[0]["lon"].ToString());
             }
```

Comment:
- Else block is redundant here.
- The conditional branch may be converted to a ternary operator.
- `DeserializeObject` may return null. If that happens, the `data.Count` will fail with NullRef Exception.

Reply:
done